### ISSUE TITLE
BaseService.deleteData methods raise exception when deleting community widget

### ISSUE THREAD

**Issue by sjleis** [OpenNTF/SocialSDK#1623]
I'm currently writing my own classes and methods to add, update, and delete community widgets via existing SBT Java code, and am having trouble with the deletion code because the SBT raises an exception even when a widget is successfully deleted.
I'm using the 1.1.0 release of the SBT.

I construct a URL like this in my code, based on Connections 4.5 API documentation:
https://connectionsServer/communities/service/atom/community/widgets?communityUuid=ab0e1737-8308-4fe2-a7aa-2b1bf6f11cd7&widgetInstanceId=W5a707cd3e97f_40aa_a0fb_d6609e216884

Then I call this SBT method to send a DELETE request to the URL, with the second and third arguments being null: `com.ibm.sbt.services.client.base.BaseService.deleteData(String,Map,String)`

Successful widget deletion causes this method to raise a `ClientServicesException`, with the lowest-level cause being `org.xml.sax.SAXParseException: Premature end of file.`

This isn't a problem with all deletions. I tested deleting a community with the SBT, and that is fine.

To investigate further, I testing deleting first a widget, then a community, using the RESTClient add-on in Firefox, and checked the response headers.
Both deletions correctly return `Content-Length: 0`, but they differ in the `Content-Type` header.
Deleting a widget returns the header `Content-Type: application/atom+xml;charset=UTF-8`, and it seems like this might be what causes the SBT exception.

To resolve this, I think the `com.ibm.sbt.services.client.Response` constructor should check the `Content-Length` response header, and only call `parseContent()` if that header's value is greater than 0.

Here's the full stack trace from when my code deletes a community widget:
```
[Fatal Error] :-1:-1: Premature end of file.
com.ibm.sbt.services.client.ClientServicesException: 
	at com.ibm.sbt.services.client.Response.parseContent(Response.java:89)
	at com.ibm.sbt.services.client.Response.<init>(Response.java:53)
	at com.ibm.sbt.services.client.ClientService.processResponse(ClientService.java:1253)
	at com.ibm.sbt.services.client.ClientService._xhr(ClientService.java:1159)
	at com.ibm.sbt.services.client.ClientService.execRequest(ClientService.java:1124)
	at com.ibm.sbt.services.client.ClientService.xhr(ClientService.java:1093)
	at com.ibm.sbt.services.client.ClientService.delete(ClientService.java:1056)
	at com.ibm.sbt.services.client.ClientService.delete(ClientService.java:1051)
	at com.ibm.sbt.services.client.base.BaseService.deleteData(BaseService.java:334)
	at com.ibm.sbt.services.client.base.BaseService.deleteData(BaseService.java:315)
	at com.isw.sbt.services.client.connections.communities.CommunityService.deleteData(CommunityService.java:151)
	at com.isw.sbt.services.client.connections.communities.CommunityService.deleteWidget(CommunityService.java:162)
	at com.isw.sbt.services.client.connections.communities.Widget.remove(Widget.java:199)
	at com.isw.connections.DominoMigration.processProfile(DominoMigration.java:360)
	at com.isw.connections.DominoMigration.runMigration(DominoMigration.java:463)
	at JavaAgent.NotesMain(JavaAgent.java:18)
	at lotus.domino.AgentBase.runNotes(Unknown Source)
	at lotus.domino.NotesThread.run(Unknown Source)
Caused by: java.io.IOException
	at com.ibm.sbt.services.client.ClientService$HandlerXml.parseContent(ClientService.java:856)
	at com.ibm.sbt.services.client.Response.parseContent(Response.java:86)
	... 17 more
Caused by: com.ibm.commons.xml.XMLException: Error parsing XML stream
	at com.ibm.commons.xml.drivers.XercesDriver.parse(XercesDriver.java:67)
	at com.ibm.commons.xml.DOMUtil.createDocument(DOMUtil.java:195)
	at com.ibm.commons.xml.DOMUtil.createDocument(DOMUtil.java:219)
	at com.ibm.sbt.services.client.ClientService$HandlerXml.parseContent(ClientService.java:854)
	... 18 more
Caused by: org.xml.sax.SAXParseException: Premature end of file.
	at org.apache.xerces.parsers.DOMParser.parse(Unknown Source)
	at org.apache.xerces.jaxp.DocumentBuilderImpl.parse(Unknown Source)
	at javax.xml.parsers.DocumentBuilder.parse(Unknown Source)
	at com.ibm.commons.xml.drivers.XercesDriver.parse(XercesDriver.java:58)
	... 21 more
```

**Comment by zipwiz** [comment]
The last activity has happened quite a while ago. Are there some recent news on this topic? Connections has moved to 5.0, and SBT has moved to 1.1.4 meanwhile...

**Comment by sjleis** [comment]
@zipwiz - I haven't tested this with any more recent SBT version than what was the current release at the time I raised this issue, and also haven't tried to fix the issue myself. I just worked around it by catching and ignoring the exception in the case of deleting widgets.
I haven't needed, or had time, to investigate further since finishing the project for which I needed this functionality in January.
